hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -173,6 +173,114 @@ def hpat_pandas_series_shape_impl(self):
     return hpat_pandas_series_shape_impl
 
 
+@overload_method(SeriesType, 'nsmallest')
+def hpat_pandas_series_nsmallest(self, n=5, keep='first'):
+    """
+    Pandas Series method :meth:`pandas.Series.nsmallest` implementation.
+
+    .. only:: developer
+       Test: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_nsmallest*
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        input series
+    n: :obj:`int`, default 5
+        Return this many ascending sorted values.
+    keep: :obj:`str`, default 'first'
+        When there are duplicate values that cannot all fit in a Series of n elements:
+        first : return the first n occurrences in order of appearance.
+        last : return the last n occurrences in reverse order of appearance.
+        all : keep all occurrences. This can result in a Series of size larger than n.
+        *unsupported*
+
+    Returns
+    -------
+    :obj:`scalar`
+         returns :obj:`series`
+    """
+
+    _func_name = 'Method nsmallest().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not isinstance(n, (types.Omitted, int, types.Integer)):
+        raise TypingError('{} The object must be an integer. Given n: {}'.format(_func_name, n))
+
+    if not isinstance(keep, (types.Omitted, str, types.UnicodeType)):
+        raise TypingError('{} The object must be an unicode. Given keep: {}'.format(_func_name, keep))
```

Comment:
In this case if to pass explicitly parameter `keep`=`'first'` as argument the exception will be raised, because `keep` will be equal to `unicode_type` and the condition will be `False`.

Reply:
@densmirn True, but why do this if this parameter is unsupported? If it's unsupported then the user should not provide any value for it (and hence only default value can be checked).